contrib/gofiber/fiber.v2/fiber.go

```diff
@@ -48,16 +49,23 @@ func Middleware(opts ...Option) func(c *fiber.Ctx) error {
 		// pass the span through the request UserContext
 		c.SetUserContext(ctx)
 
-		resourceName := c.Path()
-		if resourceName == "" {
-			resourceName = "unknown"
-		}
-		resourceName = c.Method() + " " + resourceName
-		span.SetTag(ext.ResourceName, resourceName)
-
 		// pass the execution down the line
 		err := c.Next()
 
+		// Setting the route path as resource name
+		span.SetTag(ext.ResourceName, cfg.resourceNamer(c))
+
+		if cfg.pathParams {
+			route := c.Route()
+
+			for _, parameter := range route.Params {
+				span.SetTag(
+					fmt.Sprint("http.path.", parameter),
+					c.Params(parameter),
+				)
+			}
+		}
```

Comment:
I feel like this part should be treated as a separate discussion and PR. We've not really standardised a way to do this, and I wouldn't want the discussion here to drag out the actual fix which is more important: using a route instead of the full path. Can you please remove this part? We can maybe discuss it separately if you still like.

Reply:
You are right - if there is not consensus on the way this should be accomplished, then let's discuss it in a its own PR. For now I cherry-picked this change to another branch for future discussion.

Reply:
Thanks for understanding. Happy to discuss and eventually add this if you think it's valuable.